### PR TITLE
Add (#851): Solr IndexerBolt basic test / Refactor Solr tests

### PR DESCRIPTION
**Pull request by mvolikas**

### CODE REVIEW

**Comment by jnioche** [comment]
Thanks @mvolikas 
StatusBoltTest.java and IndexerBoltTest.java share a lot of code, what about having them extend an abstract class where all that code would be put in?
We need the usual license headers in all files.
Is the content of _resources/cores_ the same as what is in _cores_? Instead of having duplicates, could we have only one copy or use links?

File: external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
+package org.apache.stormcrawler.solr.persistence;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.*;
```
Comment (rzo1) [inline external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:9]: Can we avoid * imports?

File: external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.tuple.Tuple;
+import org.apache.stormcrawler.Metadata;
+import org.apache.stormcrawler.TestOutputCollector;
+import org.apache.stormcrawler.TestUtil;
+import org.apache.stormcrawler.indexing.AbstractIndexerBolt;
+import org.apache.stormcrawler.solr.bolt.IndexerBolt;
+import org.junit.*;
```
Comment (rzo1) [inline external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:17]: Can we avoid * imports?

File: external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
+package org.apache.stormcrawler.solr.persistence;
```
Comment (rzo1) [inline external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:1]: Requires a license header

File: external/solr/src/test/resources/cores/docs/conf/schema.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/docs/conf/schema.xml:1]: Requires a license header. You can just run with the `rat` profile locally, so you see all the complains ;-)

File: external/solr/src/test/resources/cores/docs/conf/solrconfig.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<config>
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/docs/conf/solrconfig.xml:2]: Header

File: external/solr/src/test/resources/cores/docs/core.properties [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#Written by CorePropertiesLocator
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/docs/core.properties:1]: Header

File: external/solr/src/test/resources/cores/metrics/conf/schema.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/metrics/conf/schema.xml:1]: Header

File: external/solr/src/test/resources/cores/metrics/conf/solrconfig.xml [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<config>
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/metrics/conf/solrconfig.xml:2]: Header

File: external/solr/src/test/resources/cores/metrics/core.properties [hunk 9]
```diff
@@ -0,0 +1,6 @@
+#Written by CorePropertiesLocator
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/metrics/core.properties:1]: Header

File: external/solr/src/test/resources/cores/solr.xml [hunk 10]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/solr.xml:1]: Header

File: external/solr/src/test/resources/cores/status/core.properties [hunk 11]
```diff
@@ -0,0 +1,6 @@
+#Written by CorePropertiesLocator
```
Comment (rzo1) [inline external/solr/src/test/resources/cores/status/core.properties:1]: Header

**Review by rzo1 — COMMENTED** [review]
Might be worth to extract an abstract class for common code fragements.

**Comment by mvolikas** [comment]
Thank you both for the comments! I removed the duplicate docs and added a check to ensure the test document is inside the Solr core. Please have a look at the changes and let me know if anything can be improved :)

File: external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java [hunk 12]
```diff
@@ -0,0 +1,123 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more contributor license agreements. See the NOTICE
```
Comment (jnioche) [inline external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:2]: This is not the right header I'm afraid.  It should not contain any reference to DigitalPebble at all.
Please have a look at the Java classes in the main branch

**Review by jnioche — CHANGES_REQUESTED** [review]

File: external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java (hunk 12 above)
Comment (mvolikas) [reply external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java]: Sorry about that. I updated the headers - the headers in branch 851 seem to be the old ones.

**Review by jnioche — APPROVED** [review]
LGTM!

**Review by rzo1 — APPROVED** [review]

**Comment by rzo1** [comment]
Thx @mvolikas

**Pull request merged**